src/deepness/processing/map_processor/map_processing_result.py

```diff
@@ -2,22 +2,30 @@
 """
 
 
+from typing import Callable
+
+
 class MapProcessingResult:
     """
     Base class for signaling finished processing result
     """
 
-    def __init__(self, message: str):
+    def __init__(self, message: str, gui_delegate: Callable | None = None):
+        """
+        :param message: message to be shown to the user
+        :param gui_delegate: function to be called in GUI thread, as it is not safe to call GUI functions from other threads
+        """
         self.message = message
+        self.gui_delegate = gui_delegate
 
 
 class MapProcessingResultSuccess(MapProcessingResult):
     """
     Processing result on success
     """
 
-    def __init__(self, message: str = ''):
-        super().__init__(message)
+    def __init__(self, message: str = '', gui_delegate: Callable | None = None):
```

Comment:
The `|` operand is not available in older Python. I think we should replace it with `Optional` to keep compatibility.